networkx/algorithms/bipartite/matching.py

```diff
@@ -183,14 +184,18 @@ def eppstein_matching(G):
     hopcroft_karp_matching
 
     """
+    # Due to its original implementation, a directed graph is needed
+    # so that the two sets of bipartite nodes can be distinguished
+    Di_G = nx.DiGraph()
+    for node in bipartite_sets(G)[0]:
+        Di_G.add_edges_from({(node, v) for v in G[node]})
```

Comment:
So is this creating a directed graph in which the edges are all directed from nodes in the left set to nodes in the right set? If that is the case, you could just do.

    left = bipartite_sets(G)[0]
    G = nx.DiGraph(G.edges(left))